### ISSUE TITLE
After Successful installation of SMRT Link 3.1.0: Problems running HGAP 4

### ISSUE THREAD

**Issue by MichaelsGITIGIT** [PacificBiosciences/SMRT-Link#31]
We do not have Sequel system but got SMRT Link form PacBio support since we want to try the new tools incorporated there. We got the following version as a zip: smrtlink_3.1.0.180439.zip

I deployed it on Debian 8.2. I was able to install everything according to the Installation guide for v3.1.0. Also, the checking steps with `import-canneddata` and `run-sat-services` worked without error.
I was also able to convert RS II data to bam format.

Starting Google Chrome and using the GUI also works like a charm...

**When I start a job (HGAP 4) it runs for a while, until I get this error log:**
`[ERROR] 2016-06-14 13:24:17,650Z [pbsmrtpipe.tools.runner run_task 261] Failed task falcon_ns.tasks.task_falcon0_build_rdb exit code 1 in 27.63 sec (See file '/opt/pacbio/userdata/jobs_root/000/000018/tasks/falcon_ns.tasks.task_falcon0_build_rdb-0/stderr'.)
Extracted from stderr
cat raw_reads.db | awk '$1 == "blocks" {print $3}'
++ cat raw_reads.db
++ awk '$1 == "blocks" {print $3}'
awk: symbol lookup error: /opt/pacbio/install/smrtlink-fromsrc_3.1.0.180439,180439-180439-180128-180128-180128/bundles/smrttools/install/smrttools-fromsrc_3.1.0.180439/private/thirdparty/python/python_2.7.9/binwrap/../../../../thirdparty/readline/readline_6.3/lib/libreadline.so.6: undefined symbol: UP
`
This error also appears with the testdata provided by pacbio. It is not beeing caused by the data conversion from RS II as it seems.

I then modified the code which generates this bash script so that it avoids this crash.

Then this error does not occur anymore and seems to be solved. But then I get the error attached as a file

What is going wrong, have no clue :-(

[error.txt](https://github.com/PacificBiosciences/SMRT-Link/files/323426/error.txt)

**Comment by pb-cdunn** [comment]
> I then modified the code which generates this bash script so that it avoids this crash.

What modification did you make which avoided the crash? Could you post that `.sh` script?

> error.txt

    File reads.fasta is already in database raw_reads.db

To restart, you must first `rm -f raw_reads.db .raw_reads.*`. (We will make that easier eventually.)

**Comment by pb-cdunn** [comment]
Internally:
* https://bugzilla.nanofluidics.com/show_bug.cgi?id=33476

**Comment by pb-cdunn** [comment]
Also, could you tell us what these say on your system?
```sh
which awk
ldd -r $(which awk)
```

**Comment by MichaelsGITIGIT** [comment]
Thanks a lot for your replies:

`ldd -r $(which awk)` gives me:
```
linux-vdso.so.1 (0x00007ffdf0be7000)
libsigsegv.so.2 => /usr/lib/x86_64-linux-gnu/libsigsegv.so.2 (0x00007f81425ca000)
libreadline.so.6 => /lib/x86_64-linux-gnu/libreadline.so.6 (0x00007f8142380000)
libmpfr.so.4 => /usr/lib/x86_64-linux-gnu/libmpfr.so.4 (0x00007f814211d000)
libgmp.so.10 => /usr/lib/x86_64-linux-gnu/libgmp.so.10 (0x00007f8141e9a000)
libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007f8141c96000)
libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f8141995000)
libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f81415ec000)
libtinfo.so.5 => /lib/x86_64-linux-gnu/libtinfo.so.5 (0x00007f81413c2000)
/lib64/ld-linux-x86-64.so.2 (0x00007f8142a74000)
```

**Comment by MichaelsGITIGIT** [comment]
The modification I did for generating the bash-file was just to replace the following:
`count = """$(cat raw_reads.db | grep 'blocks' | sed 's/blocks = *//g')"""`
with this:
`count = """$(cat raw_reads.db | awk '$1 == "blocks" {print $3}')"""`

I did this to not call awk which seems to be connected to the problem.

**Comment by MichaelsGITIGIT** [comment]
By the way, with my modification I just mentioned in the post before, I run through now. I am not sure what exactly my 2nd problem was I posted yesterday. Might have been related to the solution you suggested yesterday (`rm -f raw_reads.db .raw_reads.*`)

**Comment by pb-cdunn** [comment]
which awk

This could be specific to Debian. I just want to be sure that we know what's going on.

> Now it works :-)
> Might have been related to the solution you suggested yesterday (rm -f raw_reads.db .raw_reads.*).

Yes.

**Comment by pb-cdunn** [comment]
Please run this:

    which awk

This could be specific to Debian. I just want to be sure that we know what's going on.

**Comment by MichaelsGITIGIT** [comment]
`which awk` gives `/usr/bin/awk`

Just in case, the awk version is:
GNU Awk 4.1.1, API: 1.1 (GNU MPFR 3.1.2-p3, GNU MP 6.0.0)

**Comment by pb-cdunn** [comment]
Thanks. Oddly, `awk` is not linked with `libreadline.so` on any other system I've seen. Anyway, the problem is solved on `master` and for `3.2.0`. I might get permission to solve it in `3.1.1` also. But I'm glad that your work-around works for you.

**Comment by MichaelsGITIGIT** [comment]
Cool, thx!
